libs/pilight/protocols/GPIO/ds18b20.c

```diff
@@ -135,14 +139,21 @@ static void *ds18b20Parse(void *param) {
 								}
 								memset(content, '\0', bytes+1);
 
-								if ((rfd = fopen(ds18b20_w1slave, "w"))) {
-									logprintf(LOG_DEBUG, "setting resolution of %s to: %d", ds18b20_w1slave, resolution);
-									if (fprintf(ffd, "%d", resolution) < 0 )
-										logprintf(LOG_ERR, "cannot set resolution of %s", ds18b20_w1slave);
-
-									fclose(ffd);
-								} else {
-									logprintf(LOG_ERR, "opening %s to set resolution failed!", ds18b20_w1slave);
+								if (uname(&unameData) != -1) {
+									char **varr = NULL;
+									int toks = explode(unameData.release, ".", &varr);
+
+									if (toks > 2 && strtol(varr[0], NULL, 10) >= 4 && strtol(varr[1], NULL, 10) >= 7) {
+										if ((rfd = fopen(ds18b20_w1slave, "w"))) {
+											logprintf(LOG_DEBUG, "setting resolution of %s to: %d", ds18b20_w1slave, resolution);
+											if (fprintf(rfd, "%d", resolution) < 0 ) {
+												logprintf(LOG_ERR, "cannot set resolution of %s", ds18b20_w1slave);
+											}
+											fclose(rfd);
+										} else {
+											logprintf(LOG_ERR, "opening %s to set resolution failed!", ds18b20_w1slave);
+										}
+									}
```

Comment:
Can there be an else here, warning the user that their settings isn't going to work?

Reply:
sure